packages/ui/atomic/template/deploy.esbuild.mjs

```diff
@@ -0,0 +1,49 @@
+import {build} from 'esbuild';
+import {join} from 'node:path';
+import {appendFileSync, readFileSync, writeFileSync, mkdirSync} from 'node:fs';
+
+function jsBundle() {
+  const entry = join('dist/components/index.js');
+  const contents = readFileSync(entry, 'utf-8');
+  const toAppend = '\ndefineCustomElements();';
+  if (!contents.includes(toAppend)) {
+    appendFileSync(entry, toAppend);
+  }
+
+  return build({
+    entryPoints: ['dist/components/index.js'],
+    bundle: true,
+    minify: true,
+    outfile: 'dist/bundle/index.js',
+    format: 'esm',
+  });
+}
+
+function cssBundle() {
+  return build({
+    entryPoints: ['src/style/index.css'],
+    outfile: 'dist/bundle/index.css',
+    bundle: true,
+    minify: true,
+  });
+}
+
+function html() {
+  const entry = join('src/pages/index.html');
+  const outputFile = join('dist/bundle/index.html');
+  const htmlContents = readFileSync(entry, 'utf-8');
+  const hostedPageContents =
+    /<atomic-hosted-page>((.|\n)*)<\/atomic-hosted-page>/gm
+      .exec(htmlContents)[1]
+      .trim();
```

Comment:
If we ever add props to `atomic-hosted-page`, this will need to be updated. I think we should either:
1. Change the regexp to support props (e.g., by replacing `<atomic-hosted-page>` with `<atomic-hosted-page(\s[^>]*)?>`), which is still flaky if we add a prop which can take a string that may contain `>`.
2. Add an HTML parser as a dev dependency.
  * I would prefer this.

Also, I think we should add some error handing in case a confused implementer removes `atomic-hosted-page` or has multiple.

Reply:
Nothing will be added here it's not a functional component, just to indicate what to parse in order to deploy that piece of html

https://github.com/coveo/cli/blob/KIT-2269/packages/ui/atomic/template/src/pages/index.html.hbs#L23